src/models/CommonModel.ts

```diff
@@ -57,10 +57,10 @@ export class CommonModel extends CommonSchema<CommonModel> {
    * 
    * @param types which types we should try and add to the existing output
    */
-  addToTypes(types: string[] | string) {
+  addTypes(types: string[] | string) {
     if (Array.isArray(types)) {
       types.forEach((value) => {
-        this.addToTypes(value);
+        this.addTypes(value);
```

Comment:
I highly recommend you to stop doing mutation. For this project it may too late.

Reply:
I don't see the issue as it is a recursive call? 🤔 We have the unit tests in place to ensure that mutation beyond what a function should do does not change.